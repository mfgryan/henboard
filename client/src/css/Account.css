#Account {
    text-align: left;
}

#Account input {
    margin-top: 6px;
}

.AccountField {
    margin-bottom: 10px
} 

.AccountButton {
    text-align: right;
    margin-top: 15px
} 
