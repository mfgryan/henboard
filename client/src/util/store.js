// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js"; // eslint-disable-line no-unused-vars
import rootReducer from "../reducers/reducers.js";

const store = createStore(rootReducer, applyMiddleware(validate, logger));

export { store };
