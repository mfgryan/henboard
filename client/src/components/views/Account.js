// react dep
import React from "react";
import { connect } from "react-redux";
import { login } from "../../actions/user.js";

// component dep
import App from "../App.js";
import { Col, Row, Button } from "react-bootstrap";

// style dep
import "../../css/Account.css";

export const Account = ({ login, user }) => {
    return (
        <App>
            <div id="Account">
                <Row>
                    <Col className="AccountField" md={4} mdOffset={4}>
                        <h3>Login</h3>
                    </Col>
                </Row>
                <Row>
                    <Col md={2} mdOffset={4}>
                        <h4>Email</h4>
                    </Col>
                    <Col md={4}>
                        <input type="text" />
                    </Col>
                </Row>
                <Row>
                    <Col md={2} mdOffset={4}>
                        <h4>Password</h4>
                    </Col>
                    <Col md={4}>
                        <input type="password"/>
                    </Col>
                </Row>
                <Row>
                    <Col md={2} mdOffset={6} className="AccountButton" >
                        <Button onClick={(user) => login({email: "bob", password: "123"})} >Submit</Button>
                    </Col>
                </Row>
            </div>
        </App>
    );
};

const mapStateToProps = state => {
    return {
        user: {
            name: "bob", 
            email: "bob@gmail.com"
        } 
    };
};

const mapDispatchToProps = dispatch => {
    return {
        login: (user) => {
            dispatch(login(user));
        }  
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Account);
