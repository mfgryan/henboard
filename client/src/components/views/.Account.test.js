import React from "react";
import Account from "./Account";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("Account renders correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <Account />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
