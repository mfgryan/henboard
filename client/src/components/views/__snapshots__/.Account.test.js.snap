exports[`test Account renders correctly 1`] = `
<div
  className="container">
  <div
    className="Header row">
    <div
      className="col-md-4" />
    <div
      className="HeaderDropdownCol col-md-4 col-md-offset-8">
      <div
        className="dropdown btn-group">
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="HeaderDropdownMenu dropdown-toggle btn btn-default"
          disabled={false}
          id="HeaderMenu"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          type="button">
          â‰¡
        </button>
        <ul
          aria-labelledby="HeaderMenu"
          className="dropdown-menu"
          role="menu">
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/account"
              onClick={[Function]}>
              Sign In
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/home"
              onClick={[Function]}>
              home
            </a>
          </li>
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/backlog"
              onClick={[Function]}>
              backlog
            </a>
          </li>
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/planning"
              onClick={[Function]}>
              planning
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              undo 
              <small>
                Ctrl-left
              </small>
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              redo 
              <small>
                Ctrl-right
              </small>
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/settings"
              onClick={[Function]}>
              settings
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    id="Account">
    <div
      className="row">
      <div
        className="AccountField col-md-4 col-md-offset-4">
        <h3>
          Login
        </h3>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-md-2 col-md-offset-4">
        <h4>
          Email
        </h4>
      </div>
      <div
        className="col-md-4">
        <input
          type="text" />
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-md-2 col-md-offset-4">
        <h4>
          Password
        </h4>
      </div>
      <div
        className="col-md-4">
        <input
          type="password" />
      </div>
    </div>
    <div
      className="row">
      <div
        className="AccountButton col-md-2 col-md-offset-6">
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button">
          Submit
        </button>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-12"
      id="footer">
      <i>
        Henboard 2017
      </i>
    </div>
  </div>
</div>
`;
